src/components/TodoItem.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  deleteTodo: (id: number) => Promise<void>;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo, deleteTodo }) => {
+  const [loadingTodo, setLoadingTodo] = useState<Todo | null>(null);
+  const { id, title, completed } = todo;
+
+  return (
+    <div data-cy="Todo" className={cn('todo', { completed: todo.completed })}>
+      {/* eslint-disable jsx-a11y/label-has-associated-control */}
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+        />
+      </label>
+
+      <span data-cy="TodoTitle" className="todo__title">
+        {title}
+      </span>
+      <button
+        type="button"
+        className="todo__remove"
+        data-cy="TodoDelete"
+        onClick={() => {
+          deleteTodo(id);
+          setLoadingTodo(todo);
```

Comment:
Move this logic from jsx to the helper variable and use it here

Reply:
Not fixed

![image](https://github.com/user-attachments/assets/09ef0a1f-39b0-42d7-949f-d6485eb7c6fa)

please, fix it in the next stage task:
```suggestion
        onClick={handleDelete}
```
create handle delete function above